### PR TITLE
Update zlib-1.2.12-SPDX2TV.spdx

### PR DESCRIPTION
**Pull request by SpessartZiegler**
In the "zlib_how.html" there is a description how to use the example zpipe.c.
In the header of zpipe.c there is a comment with "provided to the public domain". But does it mean the description is also under public domain? For me not. If the pull request would be accepted then the "Copyright (c) 2004, 2005 by Mark Adler" must be added.

### CODE REVIEW

**Review by OliverFendt — COMMENTED** [review]
@SpessartZiegler the tag 

> LicenseInfoInFile: LicenseRef-Public-domain
LicenseInfoInFile: LicenseRef-LicenseRef-scancode-public-domain

contains the scanner matches. Meaning that the license scanners "found" license matches in the file.

The tag 
> LicenseConcluded: NOASSERTION

contains the result of the review done by a person reviewing the scanner results. Here the reviewer decided that the scanner matches do not reflect the license of this file. This is mapped to NOASSERTION in this case.

In a nutshell this means the scanners identified "some public domain statement" and the reviewer decided that these statements are not the license expression of this file. Thus the information in the SPDX file is correct and I will not merge the PR

**Comment by pombredanne** [comment]
FWIW, if you run scancode-toolkit with the --license-text option you can see the exact captured text that was matched. Here this is `Not copyrighted -- provided to the public domain` that is matched from this rule https://github.com/nexB/scancode-toolkit/blob/develop/src/licensedcode/data/rules/public-domain_219.RULE  on line 23 at https://github.com/madler/zlib/blob/v1.2.12/examples/zlib_how.html#L23

For context the file has this text 
```
 20 Without further adieu, here is the program <a href="zpipe.c"><tt>zpipe.c</tt></a>:
 21 <pre><b>
 22 /* zpipe.c: example of proper use of zlib's inflate() and deflate()
 23    Not copyrighted -- provided to the public domain
 24    Version 1.4  11 December 2005  Mark Adler */
```

The file contains the code of zpipe.c embedded in HTML and the detection is correct alright. If it were not it would be a bug alright to fix in ScanCode!

You can see the same file with the same license, but not embedded in the HTML doc at https://github.com/madler/zlib/blob/v1.2.12/examples/zpipe.c 

But the HTML file also contains this footer: "Copyright (c) 2004, 2005 by Mark Adler"
and since the overall license of zlib applies, a possible conclusion is to say that https://github.com/madler/zlib/blob/v1.2.12/examples/zlib_how.html is effectively under some zlib license and contains public-domain code

We could ask @madler to confirm too!

Dear Marc:
Thank you for zlib!
It would be super gentle of you if you could clarify the license of the zlib/examples/ directory

It may seem minor and futile to you, but your code is so widely used that a lot of people may be scratching their head about this. 

Is https://github.com/madler/zlib/blob/v1.2.12/examples/zlib_how.html under the zlib license or a public domain license? It contains both your copyright notice at the bottom and pubcli-domain dedication corresponding to https://github.com/madler/zlib/blob/v1.2.12/examples/zpipe.c

**Comment by OliverFendt** [comment]
@pombredanne Yes that is a very good idea to ask Mark Adler

**Comment by madler** [comment]
The zlib source code, bolded, in http://zlib.net/zlib_how.html is not copyrighted. As noted, that same same source code with the same non-copyright can be found in the zlib distribution: https://github.com/madler/zlib/blob/master/examples/zpipe.c . The unbolded description interspersed between the lines of code on that web page _is_ copyrighted, as noted at the bottom of the web page.

A copyrighted web page contains a copy of non-copyrighted material. What is the issue?

**Comment by OliverFendt** [comment]
@madler thank you very much for feedback. In general there is no problem having uncopyrighted things included in copyrighted stuff. The question I still have is about the file https://github.com/madler/zlib/blob/v1.2.12/examples/zlib_how.html do the copyrighted material (the description without the non copyright code fragments) go under the Zlib license?

**Comment by madler** [comment]
No. Nothing in zlib_how.html refers to the zlib license.

Do you want it to? I suppose I could put it under the zlib license. Though I'm not sure HTML counts as "software".

**Comment by OliverFendt** [comment]
Ok I see, the thing is that without a license attatched I think that the use of the file is very limited since the terms of use are unknown. So it would be very useful if this would be licensed. In case the Zlib license seems not to be appropriate [CC-BY-4.0](https://creativecommons.org/licenses/by/4.0/deed.en) is a license which is frequently used for documentation.

**Comment by madler** [comment]
I added a license to the web page, as well as to the corresponding file in examples with the commit https://github.com/madler/zlib/commit/a9b4c1de37bb1962048a6a83658479c3902ca9f8 .

**Comment by OliverFendt** [comment]
@madler Thank you very much for this. My (hopefully) last questions are: does CC-BY-ND-4.0 also apply to [version 1.2.12 of zlib_how.html](https://github.com/madler/zlib/blob/v1.2.12/examples/zlib_how.html)
Since CC-BY-ND-4.0 requires an acknowledgment (in CC terms "appropriate credit") which shall inculde: 
 - name of the creator  
 - copyright notice  
 - license notice  
 - disclaimer  
 - link to the material 

Is the following fine:

> This work contains material copyrighted by Marc Adler (https://en.wikipedia.org/wiki/Mark_Adler). This material is licensed under CC-BY-ND-4.0, you can find the text of CC-BY-ND-4.0 at: https://creativecommons.org/licenses/by-nd/4.0/legalcode

> For more information please visit: https://github.com/madler/zlib

> Disclaimer of Warranties and Limitation of Liability.

> Unless otherwise separately undertaken by the Licensor, to the extent possible, the Licensor offers the Licensed Material as-is and as-available, and makes no representations or warranties of any kind concerning the Licensed Material, whether express, implied, statutory, or other. This includes, without limitation, warranties of title, merchantability, fitness for a particular purpose, non-infringement, absence of latent or other defects, accuracy, or the presence or absence of errors, whether or not known or discoverable. Where disclaimers of warranties are not allowed in full or in part, this disclaimer may not apply to You.

> To the extent possible, in no event will the Licensor be liable to You on any legal theory (including, without limitation, negligence) or otherwise for any direct, special, indirect, incidental, consequential, punitive, exemplary, or other losses, costs, expenses, or damages arising out of this Public License or use of the Licensed Material, even if the Licensor has been advised of the possibility of such losses, costs, expenses, or damages. Where a limitation of liability is not allowed in full or in part, this limitation may not apply to You.

**Comment by sschuberth** [comment]
> Here the reviewer decided that the scanner matches do not reflect the license of this file. This is mapped to NOASSERTION in this case.

Unrelated to the PR as such, but shouldn't `NONE` be used instead to indicate that none of the licenses found is a true-positive?

**Comment by OliverFendt** [comment]
I close the PR. I will consider the information in further analysis
